site/docs/references/pseudo-classes.md

```diff
@@ -38,12 +38,248 @@ To define custom pseudo-classes, or states, for a simple selector, you tell **St
+
+```css
+/* defining state */
+.token{
+    -st-states: fieldName(string);
+}
+
+/* customize fields with fieldName email */
+.token:fieldName(email){
+    color:lightBlue;
+}
+
+/* customize fields with fieldName that starts with user_ */
+.token:fieldName(^user_){
+    color:blue;
+}
+
+/* customize fields with fieldName that ends with _id */
+.token:fieldName($_id){
+    color:gray;
+}
+
+/* customize fields with fieldName that includes error */
+.token:fieldName(*error){
+    color:red;
+}
+
+
+/* using includes with "not" operator */
+.token:fieldName(!*error){
+    border:1px solid green;
+}
+```
+
+```css
+/* CSS output*/
+.Example1__root[data-Example1-fieldName="email"] { color: lightBlue; }
+.Example1__root[data-Example1-fieldName^="user_"] { color: blue; }
+.Example1__root[data-Example1-fieldName$="_id"] { color: gray; }
+.Example1__root[data-Example1-fieldName*="error"] { color: red; }
+.Example1__root:not([data-Example1-fieldName*="error"]) { color: red; }
+
+```
+
+
+#### Number example
+
+```css
+/* stateWithNumberParam.st.css */
+.cell{
+    -st-states: column(number);
+}
+
+/* customize fields with at column 1 */
+.cell:column(1){
+    color:lightBlue;
+}
+
+/* customize using css number matching ( FUTURE ) */
+.cell:column(2n+1){
+    color:green;
+}
+
+/* customize using css odd/even keywords ( FUTURE ) */
+.cell:column(even){
+    color:grey;
+}
+
+/* customize column greater then 3 ( FUTURE ) */
+.cell:column(>3){
+    background:blue;
+}
+
+/* customize column lesser then 2 ( FUTURE ) */
+.cell:column(<2){
+    background:red;
+}
+
+```
+
+
+```css
+/* CSS output*/
+.Example1__root[data-Example1-column="1"] { color: lightBlue; }
+.Example1__root[data-Example1-column$modula="2|1"] { color: green; }
+.Example1__root[data-Example1-column$modula="2|0"] { color: grey; }
+.Example1__root[data-Example1-column$time-line="^a"] { color: blue; }
+.Example1__root[data-Example1-column$time-line="^abc"] { background: blue; }
+.Example1__root[data-Example1-column$n+1="^abc"] { background: blue; }
+.Example1__root:not([data-Example1-column$time-line="*c"]) { background: red; }
+
+```
+
+*all the future features described above are implemented using static processing.*
+
+*because of that you will have to declare them at build time in order to use them with dynamic css*
+
+```css
+.cell{
+    -st-states: column(number);
+    -st-states: done(number<0,100>);
+}
+
+
+.cell:column(2n+x):column(3n+x):column(4n+x){
+    /*will let stylable know you're using the css modula syntax at runtime with: 
+    2n+x,3n+x or 4n+x, stylable runtime will now know to add the right data attributes*/
+}
+
+.cell:column(x){
+    /* will let stylable know your gonna use the number current limits, assumes a stepValue of 1, minValue of 0, maxValue of 10 if not specified*/
+}
+
+
+.cell:column(5x){
+    /* will let stylable know your gonna use the number current limits, with step value 5,
+    minValue of 0, maxValue of 10*/
+}
+
+```
+
+#### Boolean example
```

Comment:
Make this first example, make clear it is the default. 
If it is possible to give an explicit Boolean parameter, give an example.